### PR TITLE
[12.0][MIG] hr_timesheet_sheet_period

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Migration of module hr_timesheet_sheet_period to v12

### CODE REVIEW

File: hr_timesheet_sheet_period/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': "HR Timesheet Sheet based on Payroll Period",
+    'version': '12.0.1.0.0',
+    'category': 'Human Resources',
+    "author": "Eficent Business and IT Consulting Services S.L., "
+              "Serpent Consulting Services Pvt. Ltd., "
+              "Odoo Community Association (OCA)",
+    'website': 'http://www.eficent.com',
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_period/__manifest__.py]: @ageficent website URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

File: hr_timesheet_sheet_period/models/hr_timesheet_sheet.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+# Copyright 2016-17 Eficent Business and IT Consulting Services S.L.
+# Copyright 2016-17 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError as UserError
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_period/models/hr_timesheet_sheet.py]: @ageficent here direct use `ValidationError` , no need to `as`

File: hr_timesheet_sheet_period/models/hr_timesheet_sheet.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+class HrTimesheetSheet(models.Model):
+    _inherit = "hr_timesheet.sheet"
+
+    @api.model
+    def _default_date_start(self):
+        res = super(HrTimesheetSheet, self)._default_date_start()
+        period = self._get_current_pay_period()
+        if period:
+            return period.date_start
+        else:
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_period/models/hr_timesheet_sheet.py]: @ageficent no need to `else`
```
if period:
    return period.date_start
return res
```

File: hr_timesheet_sheet_period/models/hr_timesheet_sheet.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            return res
+
+    @api.model
+    def _default_date_end(self):
+        res = super(HrTimesheetSheet, self)._default_date_end()
+        period = self._get_current_pay_period()
+        if period:
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_period/models/hr_timesheet_sheet.py]: same here

File: hr_timesheet_sheet_period/models/hr_timesheet_sheet.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        if self.hr_period_id:
+            self.date_start = self.hr_period_id.date_start
+            self.date_end = self.hr_period_id.date_end
+            self.name = self.hr_period_id.name
+
+    @api.model
+    def _get_current_pay_period(self):
+        period_obj = self.env['hr.period']
+        contract_obj = self.env['hr.contract']
+        date_today = datetime.today().strftime('%Y-%m-%d')
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_period/models/hr_timesheet_sheet.py]: @ageficent here use server date

File: hr_timesheet_sheet_period/views/hr_timesheet_sheet_view.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016-17 Eficent Business and IT Consulting Services S.L.
+     Serpent Consulting Services Pvt. Ltd.
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_period/views/hr_timesheet_sheet_view.xml]: `Copyright 2016-17 Serpent Consulting Services Pvt. Ltd.`

File: hr_timesheet_sheet_period/tests/test_hr_timesheet_sheet_period.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        timesheet_sheet = self.timeheet_model.create({
+            'employee_id': self.employee.id
+        })
+        return timesheet_sheet
+
+    def test_defaults(self):
+        self.create_fiscal_year()
+        hr_timesheet = self.create_hr_timesheet_sheet()
+        self.assertEqual(
+            hr_timesheet.date_end.strftime('%Y-%m-%d'),
```
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_period/tests/test_hr_timesheet_sheet_period.py]: same as here and also check other

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: hr_timesheet_sheet_period/models/hr_timesheet_sheet.py (hunk 3 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet_period/models/hr_timesheet_sheet.py]: or even
```
return period.date_start if period else res
```
or
```
return period and period.date_start or res
```

File: hr_timesheet_sheet_period/tests/test_hr_timesheet_sheet_period.py (hunk 7 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet_period/tests/test_hr_timesheet_sheet_period.py]: @aheficent is there real need to convert to string?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: hr_timesheet_sheet_period/tests/test_hr_timesheet_sheet_period.py (hunk 7 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet_period/tests/test_hr_timesheet_sheet_period.py]: wrong tag, should've been @ageficent

**Review by MiquelRForgeFlow — APPROVED** [review]
Comments attended

**Review by alexey-pelykh — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@nikul-serpentcs please re-review

**Pull request merged**